miniserver/routes/middleware.js

```diff
@@ -0,0 +1,120 @@
+      if (smsParams[1].substr(0, 2) === '0x') {
+        if (ethers.utils.isAddress(smsParams[1])) {
+          funder.address = smsParams[1]
+        } else {
+          response.message(`error: invalid funder address ${smsParams[1]}. example request "p 0x8ba1f109551bd432803012645ac136ddd64dba72 0.1"`)
+          return res.send(response.toString())
+        }
+      } else {
+        // set country and look up user address from Phone Number
+        let funderPhone = smsParams[1]
+        if (funderPhone.substr(0, 1) !== '+') {
+          const { countryIso3 } = phone(requestor.phone)
+          const { isValid, phoneNumber } = phone(funderPhone, countryIso3)
+          funderPhone = phoneNumber
+          if (!isValid) {
+            response.message(`error: invalid recipient phone number ${smsParams[1]}. example request "p +14158401999 0.1"`)
+            return res.send(response.toString())
+          }
+        }
+        Logger.log(`funderPhone: ${funderPhone}`)
+        funder = await User.findByPhone({ phone: funderPhone })
+        if (!funder?.address) {
+          response.message(`error: funders phone number is not a registered wallet: ${smsParams[1]}. example request "p +14158401999 0.1"`)
+          return res.send(response.toString())
+        }
+      }
+
+      if (toNumber(smsParams[2]) > 0) {
+        amount = ethers.utils.parseEther(smsParams[2])
+      } else {
+        response.message(`error: pay request requires a valid amount': ${smsParams[2]} example request "p +14158401999 0.1"`)
+        return res.send(response.toString())
+      }
+      req.processedBody = { ...req.processedBody, command, requestor, funder, amount }
+    } else {
+      response.message('error: invalid sms command. example payment request "p +14158401999 0.1"')
+      return res.send(response.toString())
+    }
+  } catch (ex) {
+    console.error(ex)
+    const { code, error, success } = parseError(ex)
+    response.message(`An exception occured. success: ${success} code: ${code} error: ${error.substr(0, 100)} `)
+    return res.send(response.toString())
+  }
+  next()
+}
+
+const validateID = async (req, res, next) => {
+  const { phoneNumber, address } = req.processedBody
+  let u
+  if (address) {
+    if (!ethers.utils.isAddress(address)) {
+      return res.status(StatusCodes.BAD_REQUEST).json({ error: `invalid Address: ${address}` })
+    } else {
+      // find by address
+      u = await User.findByAddress({ address })
+      if (!u?.id) {
+        return res.status(StatusCodes.BAD_REQUEST).json({ error: `address does not exists: ${address}` })
+      }
+    }
+  } else {
+    //   const u = await User.findByAddress({ address })
+    u = await User.findByPhone({ phone: phoneNumber })
+    if (!u?.id) {
+      return res.status(StatusCodes.BAD_REQUEST).json({ error: `phone number not registered: ${phoneNumber}` })
+    }
+  }
+  console.log(`u: ${JSON.stringify(u)}`)
```

Comment:
need to remove - leaking user info in console

Reply:
removed 🙏